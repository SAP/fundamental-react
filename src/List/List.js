import classnamesBind from 'classnames/bind';
import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
import ListByline from './_ListByline';
import ListFooter from './_ListFooter';
import ListHeader from './_ListHeader';
import ListIcon from './_ListIcon';
import ListItem from './_ListItem';
import ListSelection from './_ListSelection';
import ListText from './_ListText';
import PropTypes from 'prop-types';
import React from 'react';
import useUniqueId from '../utils/useUniqueId';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/list.css';

const classnames = classnamesBind.bind(styles);

/** **List** and **Table** are similar as both usually contain a vertical list of data,
but lists generally contain basic data and tables tend to hold more complex data.
If the list is a complex hierarchy, it is best to use a **Tree**. */
const List = React.forwardRef(({
    children,
    className,
    cssNamespace,
    compact,
    footer,
    footerClassName,
    hasByline,
    header,
    headerClassName,
    headerStyle,
    id,
    level,
    navigation,
    noBorder,
    partialNavigation,
    selectable,
    ...props
}, ref) => {

    const listClasses = classnames(
        `${cssNamespace}-list`,
        {
            [`${cssNamespace}-list--selection`]: selectable,
            [`${cssNamespace}-list--compact`]: compact,
            [`${cssNamespace}-list--no-border`]: noBorder,
            [`${cssNamespace}-list--byline`]: hasByline,
            [`${cssNamespace}-list--navigation`]: navigation || partialNavigation,
            [`${cssNamespace}-list--navigation-indication`]: navigation || partialNavigation
        },
        className
    );

    const listId = useUniqueId(id);
    const listHeaderId = `${listId}-header`;

    return (
        <>
            {header &&
                <List.Header
                    className={headerClassName}
                    headingStyle={headerStyle}
                    id={listHeaderId}
                    level={level}>
                    {header}
                </List.Header>
            }
            <ul
                {...props}
                aria-labelledby={header ? listHeaderId : null}
                className={listClasses}
                id={`${listId}-list`}
                ref={ref}
                role={selectable ? 'listbox' : 'list'}>
                { React.Children.map(children, child => React.cloneElement(child, { hasByline, navigation, partialNavigation })) }
            </ul>
            {footer &&
                <List.Footer className={footerClassName}>
                    {footer}
                </List.Footer>
            }
        </>
    );
});

List.displayName = 'List';

List.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** to enable compact mode */
    compact: PropTypes.bool,
    /** The list footer as a String or a React node.*/
    footer: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
    /** CSS class(es) to add to the list footer */
    footerClassName: PropTypes.string,
    /** Set to **true** if any list item has a byline. */
    hasByline: PropTypes.bool,
    /** The list header as a String or a React node.*/
    header: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
    /** CSS class(es) to add to the list header */
    headerClassName: PropTypes.string,
    /** Heading style, if it should be different from the default style for the heading level. */
    headerStyle: CustomPropTypes.range(2, 6),
    /** Unique id for the list, used to associate `List.Header` as the list label for accessibility. A generated value will be used if not set.*/
    id: PropTypes.string,
    /** List header's semantic level. `<h1>` is reserved for the page title. It should not appear in components */
    level: CustomPropTypes.range(2, 6),
    /** Set to **true** if all list items are links */
    navigation: PropTypes.bool,
    /** Set to **true** to remove borders from the List component. */
    noBorder: PropTypes.bool,
    /** Set to **true** if any list item is a link, but not all */
    partialNavigation: PropTypes.bool,
    /** Set to **true** if list is an option list i.e. `List.Item` contain `List.Selection`. Do not add non-selectable list items to such lists for accessibility reasons.*/
    selectable: PropTypes.bool
};

List.defaultProps = {
    level: 4
};


List.Footer = ListFooter;
List.Header = ListHeader;
List.Item = ListItem;
List.Icon = ListIcon;
List.Text = ListText;
List.Byline = ListByline;
List.Selection = ListSelection;

export default withStyles(List);
