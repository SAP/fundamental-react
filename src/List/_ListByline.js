import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/list.css';

const classnames = classnamesBind.bind(styles);

const ListByline = ({
    className,
    children,
    cssNamespace,
    twoColumns,
    ...props
}) => {

    const ListBylineClasses = classnames(
        `${cssNamespace}-list__byline`,
        {
            [`${cssNamespace}-list__byline--2-col`]: twoColumns
        },
        className
    );


    return (
        <div {...props} className={ListBylineClasses}>
            {children}
        </div>
    );
};


ListByline.displayName = 'List.Byline';

ListByline.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** if two List.Text bylines will be passed as children. */
    twoColumns: PropTypes.bool
};

ListByline.defaultProps = {
    twoColumns: false
};

export default withStyles(ListByline);
