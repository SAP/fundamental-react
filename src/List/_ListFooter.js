import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const ListFooter = ({
    className,
    children,
    ...props
}) => {
    const ListItemClasses = classnames(
        'fd-list__footer',
        className
    );

    return (
        <li {...props} className={ListItemClasses}>
            {children}
        </li>

    );
};

ListFooter.displayName = 'List.Footer';

ListFooter.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default ListFooter;
