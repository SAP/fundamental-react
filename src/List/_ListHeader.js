import classnamesBind from 'classnames/bind';
import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
import PropTypes from 'prop-types';
import React from 'react';
import Title from '../Title/Title';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/list.css';

const classnames = classnamesBind.bind(styles);

const ListHeader = ({
    className,
    children,
    cssNamespace,
    headingStyle,
    level,
    ...props
}) => {
    const ListItemClasses = classnames(
        `${cssNamespace}-list__group-header`,
        className
    );

    return (
        <Title
            {...props}
            className={ListItemClasses}
            level={level}
            levelStyle={headingStyle}>
            {children}
        </Title>
    );
};

ListHeader.displayName = 'List.Header';

ListHeader.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Heading style, if it should be different from the default style for the heading level. */
    headingStyle: CustomPropTypes.range(2, 6),
    /** Header level. `<h1>` is reserved for the page title. It should not appear in components */
    level: CustomPropTypes.range(2, 6)
};

export default withStyles(ListHeader);
