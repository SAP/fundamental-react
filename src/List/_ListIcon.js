import classnames from 'classnames';
import { listOfIcons } from '../utils/listOfIcons';
import PropTypes from 'prop-types';
import React from 'react';

const ListIcon = ({ className, glyph, ...props }) => {
    const listIconClasses = classnames(
        'fd-list__icon',
        {
            [`sap-icon--${glyph}`]: glyph
        },
        className
    );

    return (
        <span {...props} className={listIconClasses} />
    );
};

ListIcon.displayName = 'List.Icon';

ListIcon.propTypes = {
    glyph: PropTypes.oneOf(listOfIcons).isRequired,
    className: PropTypes.string
};

export default ListIcon;
