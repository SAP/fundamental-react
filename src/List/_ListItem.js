import classnamesBind from 'classnames/bind';
import ListSelection from './_ListSelection';
import PropTypes from 'prop-types';
import withStyles from '../utils/withStyles';
import React, { useState } from 'react';
import styles from 'fundamental-styles/dist/list.css';

const classnames = classnamesBind.bind(styles);

const ListItem = ({
    action,
    buttonProps,
    className,
    children,
    cssNamespace,
    hasByline,
    onClick,
    navigation,
    partialNavigation,
    selected,
    url,
    ...props
}) => {
    const [selectedState, setSelectedState] = useState(selected || false);

    const handleClick = (e) => {
        onClick(e);
    };

    const isLink = navigation || (partialNavigation && url);

    const ListItemClasses = classnames(
        `${cssNamespace}-list__item`,
        {
            [`${cssNamespace}-list__item--link`]: isLink,
            [`${cssNamespace}-list__item--action`]: action
        },
        className
    );

    let content, selectionProps;

    if (hasByline) {
        content = (
            <div className={classnames(`${cssNamespace}-list__content`)}>
                {children}
            </div>
        );
    } else if (isLink) {
        const linkClassNames = classnames(
            `${cssNamespace}-list__link`,
            {
                [`${cssNamespace}-list__link--navigation-indicator`]: partialNavigation,
                'is-selected': selected
            }
        );

        content = (
            <a
                className={linkClassNames}
                href={url}
                tabIndex='0'>
                {children}
            </a>
        );
    } else if (action) {
        content = (
            <button {...buttonProps}
                className={classnames(`${cssNamespace}-list__title`)}
                onClick={handleClick}>
                {children}
            </button>
        );
    } else {
        content = React.Children.map(children, child => {
            if (child?.type?.displayName === ListSelection.displayName) {
                selectionProps = {
                    'role': 'option',
                    'aria-selected': selectedState
                };
                return React.cloneElement(child, {
                    selected: selected || selectedState,
                    onChange: (e, checked) => {
                        setSelectedState(checked);
                        child?.props?.onChange?.(e, checked);
                    }
                });
            }
            return child;
        });
    }

    const disableListItemOnClick = isLink || action;

    return (
        <li
            tabIndex={isLink || action ? '-1' : '0'}
            {...props}
            {...selectionProps}
            className={ListItemClasses}
            onClick={disableListItemOnClick ? null : handleClick}>
            {content}
        </li>

    );
};

ListItem.displayName = 'List.Item';

ListItem.propTypes = {
    /** Set to true if list item is a button that will trigger an action */
    action: PropTypes.bool,
    /** Props to pass to the action button */
    buttonProps: PropTypes.object,
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Internal use only */
    hasByline: PropTypes.bool,
    /** Internal use only */
    navigation: PropTypes.bool,
    /** Internal use only */
    partialNavigation: PropTypes.bool,
    /** Set to **true** if list item is currently selected (only supported for links and List.Selection) */
    selected: PropTypes.bool,
    /** URL to navigate to if list item is a link */
    url: PropTypes.string,
    /**
     * Callback function; triggered when user clicks on the list item i.e. `<li>`
     *
     *  (not supported for links as they are supposed to navigate).
     *
     * @param {SyntheticEvent} event - React's original SyntheticEvent. See https://reactjs.org/docs/events.html.
     * @returns {void}
     */
    onClick: PropTypes.func
};

ListItem.defaultProps = {
    onClick: () => {},
    selected: false
};

export default withStyles(ListItem);
