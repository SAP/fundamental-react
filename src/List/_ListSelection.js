import Checkbox from '../Forms/Checkbox';
import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import withStyles from '../utils/withStyles';
import React, { useState } from 'react';
import formItemStyles from 'fundamental-styles/dist/form-item.css';
import listStyles from 'fundamental-styles/dist/list.css';

const classnames = classnamesBind.bind({
    ...formItemStyles,
    ...listStyles
});

const ListSelection = ({
    checkBoxAriaLabel,
    className,
    children,
    cssNamespace,
    onChange,
    selected,
    ...props
}) => {
    const [selectedState, setSelectedState] = useState(selected || false);

    const ListSelectionClasses = classnames(
        `${cssNamespace}-form-item`,
        `${cssNamespace}-list__form-item`,
        className
    );

    return (
        <>
            <div {...props} className={ListSelectionClasses}>
                <Checkbox
                    ariaLabel={checkBoxAriaLabel}
                    checked={selectedState}
                    inline
                    onChange={(e, checked) => {
                        setSelectedState(checked);
                        onChange(e, checked);
                    }} />
            </div>
            {children}
        </>
    );
};

ListSelection.displayName = 'List.Selection';

ListSelection.propTypes = {
    /** aria-label for the checkbox */
    checkBoxAriaLabel: PropTypes.string.isRequired,
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Boolean value controlled by parent List.Item*/
    selected: PropTypes.bool,
    /**
     * Callback function; triggered when the change event fires on the underlying HTML checkbox `<input>`
     * of the parent `<li>`.
     *
     * @param {SyntheticEvent} event - React's original SyntheticEvent. See https://reactjs.org/docs/events.html.
     * @param {Boolean} checkedState - represents the final checked state of the HTML checkbox input.
     * @returns {void}
    */
    onChange: PropTypes.func
};

export default withStyles(ListSelection);
