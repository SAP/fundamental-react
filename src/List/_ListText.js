import classnamesBind from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/list.css';

const classnames = classnamesBind.bind(styles);

const ListText = ({
    children,
    className,
    cssNamespace,
    left,
    noWrap,
    right,
    secondary,
    ...props
}) => {

    let type = 'title';

    if (secondary) {
        type = 'secondary';
    }

    const listTextClasses = classnames(
        {
            [`${cssNamespace}-list__${type}`]: !left && !right,
            [`${cssNamespace}-list__${type}--no-wrap`]: noWrap,
            [`${cssNamespace}-list__byline-left`]: left,
            [`${cssNamespace}-list__byline-right`]: right
        },
        className
    );

    return (
        <span {...props} className={listTextClasses}>
            {children}
        </span>

    );
};

ListText.displayName = 'List.Text';

ListText.propTypes = {
    /** Node(s) to render within the component */
    children: PropTypes.node,
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** if there are two bylines to left align the text. */
    left: PropTypes.bool,
    /** Text is wrapped by default, set to **true** to prevent the text from wrapping */
    noWrap: PropTypes.bool,
    /** Set to **true** if there are two bylines to right align the text. */
    right: PropTypes.bool,
    /** Set to **true** to right align the text in the list */
    secondary: PropTypes.bool
};

export default withStyles(ListText);
