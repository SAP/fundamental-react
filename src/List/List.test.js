import List from './List';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<List />', () => {
    const defaultList = (
        <List>
            <List.Header>List Header</List.Header>
            <List.Item>
                <List.Text>List Item 1</List.Text>
            </List.Item>
            <List.Item>
                <List.Text>List Item 2</List.Text>
            </List.Item>
            <List.Item>
                <List.Text>List Item 3</List.Text>
            </List.Item>
            <List.Item>
                <List.Text>List Item 4</List.Text>
            </List.Item>
            <List.Footer>List Footer</List.Footer>
        </List>
    );
    const headerList = (
        <List>
            <List.Header>List Header</List.Header>
            <List.Item>
                <List.Text>List Item 1</List.Text>
            </List.Item>
            <List.Item>
                <List.Text>List Item 2</List.Text>
            </List.Item>
            <List.Footer>List Footer</List.Footer>
        </List>
    );

    const borderlessList = (
        <List noBorder>
            <List.Item>
                <List.Text>List Item 1</List.Text>
            </List.Item>
        </List>
    );

    const compactList = (
        <List compact>
            <List.Item>
                <List.Text>List Item 1</List.Text>
            </List.Item>
        </List>
    );

    const leftIconsList = (
        <List>
            <List.Header>List Header</List.Header>
            <List.Item>
                <List.Icon glyph='cart' />
                <List.Text>List Item 1</List.Text>
            </List.Item>
            <List.Item>
                <List.Icon glyph='wrench' />
                <List.Text>List Item 2</List.Text>
            </List.Item>
            <List.Item>
                <List.Icon glyph='lightbulb' />
                <List.Text>List Item 3</List.Text>
            </List.Item>
            <List.Item>
                <List.Icon glyph='history' />
                <List.Text>List Item 4</List.Text>
            </List.Item>
            <List.Footer>List Footer</List.Footer>
        </List>
    );

    const rightIconList = (
        <List>
            <List.Header>List Header</List.Header>
            <List.Item>
                <List.Text>List Item 1</List.Text>
                <List.Icon glyph='navigation-right-arrow' />
            </List.Item>
            <List.Item>
                <List.Text>List Item 2</List.Text>
                <List.Icon glyph='navigation-right-arrow' />
            </List.Item>
            <List.Item>
                <List.Text>List Item 3</List.Text>
                <List.Icon glyph='navigation-right-arrow' />
            </List.Item>
            <List.Item>
                <List.Text>List Item 4</List.Text>
                <List.Icon glyph='navigation-right-arrow' />
            </List.Item>
            <List.Footer>List Footer</List.Footer>
        </List>
    );

    const secondaryList = (
        <List>
            <List.Header>List Header</List.Header>
            <List.Item>
                <List.Text>List Item 1</List.Text>
                <List.Text secondary>Positive</List.Text>
            </List.Item>
            <List.Item>
                <List.Text>List Item 2</List.Text>
                <List.Text secondary>Negative</List.Text>
            </List.Item>
            <List.Item>
                <List.Text>List Item 3</List.Text>
                <List.Text secondary>Positive</List.Text>
            </List.Item>
            <List.Item>
                <List.Text>List Item 4</List.Text>
                <List.Text secondary>Negative</List.Text>
            </List.Item>
            <List.Footer>List Footer</List.Footer>
        </List>
    );

    test('create list group', () => {
        // create default list
        let component = renderer.create(defaultList);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list with header + footer
        component = renderer.create(headerList);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list with no borders
        component = renderer.create(borderlessList);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list compact
        component = renderer.create(compactList);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list with left icons
        component = renderer.create(leftIconsList);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list with right icons
        component = renderer.create(rightIconList);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list with secondary titles
        component = renderer.create(secondaryList);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the List component', () => {
            const element = mount(<List data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });

    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <List ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('UL');
        expect(ref.current.className).toEqual('fd-list');
    });
});
