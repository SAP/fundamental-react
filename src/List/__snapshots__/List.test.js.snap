// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> create list group 1`] = `
<ul
  className="fd-list"
>
  <li
    className="fd-list__group-header"
  >
    List Header
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 1
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 2
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 3
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 4
    </span>
  </li>
  <li
    className="fd-list__footer"
  >
    List Footer
  </li>
</ul>
`;

exports[`<List /> create list group 2`] = `
<ul
  className="fd-list"
>
  <li
    className="fd-list__group-header"
  >
    List Header
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 1
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 2
    </span>
  </li>
  <li
    className="fd-list__footer"
  >
    List Footer
  </li>
</ul>
`;

exports[`<List /> create list group 3`] = `
<ul
  className="fd-list fd-list--no-border"
>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 1
    </span>
  </li>
</ul>
`;

exports[`<List /> create list group 4`] = `
<ul
  className="fd-list fd-list--compact"
>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 1
    </span>
  </li>
</ul>
`;

exports[`<List /> create list group 5`] = `
<ul
  className="fd-list"
>
  <li
    className="fd-list__group-header"
  >
    List Header
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__icon sap-icon--cart"
    />
    <span
      className="fd-list__title"
    >
      List Item 1
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__icon sap-icon--wrench"
    />
    <span
      className="fd-list__title"
    >
      List Item 2
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__icon sap-icon--lightbulb"
    />
    <span
      className="fd-list__title"
    >
      List Item 3
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__icon sap-icon--history"
    />
    <span
      className="fd-list__title"
    >
      List Item 4
    </span>
  </li>
  <li
    className="fd-list__footer"
  >
    List Footer
  </li>
</ul>
`;

exports[`<List /> create list group 6`] = `
<ul
  className="fd-list"
>
  <li
    className="fd-list__group-header"
  >
    List Header
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 1
    </span>
    <span
      className="fd-list__icon sap-icon--navigation-right-arrow"
    />
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 2
    </span>
    <span
      className="fd-list__icon sap-icon--navigation-right-arrow"
    />
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 3
    </span>
    <span
      className="fd-list__icon sap-icon--navigation-right-arrow"
    />
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 4
    </span>
    <span
      className="fd-list__icon sap-icon--navigation-right-arrow"
    />
  </li>
  <li
    className="fd-list__footer"
  >
    List Footer
  </li>
</ul>
`;

exports[`<List /> create list group 7`] = `
<ul
  className="fd-list"
>
  <li
    className="fd-list__group-header"
  >
    List Header
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 1
    </span>
    <span
      className="fd-list__secondary"
    >
      Positive
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 2
    </span>
    <span
      className="fd-list__secondary"
    >
      Negative
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 3
    </span>
    <span
      className="fd-list__secondary"
    >
      Positive
    </span>
  </li>
  <li
    aria-selected={false}
    className="fd-list__item"
    onClick={[Function]}
    role="option"
    tabIndex="0"
  >
    <span
      className="fd-list__title"
    >
      List Item 4
    </span>
    <span
      className="fd-list__secondary"
    >
      Negative
    </span>
  </li>
  <li
    className="fd-list__footer"
  >
    List Footer
  </li>
</ul>
`;
